### DISCUSSION TITLE
CF 'randomly' flips over in the middle of a sequence using the HL commander.

### DISCUSSION THREAD

**Discussion by EJKT** [bitcraze/discussions#174] [Q&A] [answered]
My setup is the following: a CF 2.1 with a flow deck v2. I am using the HL commander in order to make use of controllable velocity in the z direction (see [this thread](https://github.com/orgs/bitcraze/discussions/134#discussioncomment-3195492) ). On some occasions near the end or start of the sequence, the CF flips over ballistically out of nowhere. All the program (coded with the python API) is doing is making the CF go up, drop, lift back up, and then land. Another thing that I am doing, but this is in a different function that is not running during the code execution is that I am using the commander instance in combination with the HL commander - perhaps this is not a good idea.

I also noticed that on the client software when running a propeller test on the console, I get the following warnings: 

STAB: WARNING: stabilizer loop rate is off (778)
ESTKALMAN: WARNING: Kalman prediction rate low (90)

I am wondering whether they are related. But in any case, I would like to know how to solve both issues. Thanks in advance.

**Comment by marios-stam** [comment, marked as answer]
Hello,

can you provide the data from the debug tab of the cfclient while being connected to the Crazyflie in order to get a better insight into your problem?

Have you checked that the position estimation data through the cfclient seem ok? You can also try to use the cfclient command-based flight control to fly the Crazyflie or some examples as Kimberly suggested in the other thread and test if you have the same issues, if yes that means that there is probably a problem with the Crazyflie setup and not your code. 

Which function of the HL commander do you use? While using the commander, sudden changes to setpoints or asking to go to a setpoint in a very quick amount of time can lead to the saturation of the motors from the controller and result in crashes so you could try to increase the time duration of each sequence.

You can use the HL commander in combination with the Commander but you will have to be careful in the transitions and what setpoint are you demanding. Since Commander does not plan a smooth trajectory to the given setpoint, demanding aggressive movements may lead to instability and crashes. 

It is also recommended to use the flowdeck in areas where the ground is not of constant color and texture but rich in different colors and patterns. This will help the Crazyflie have a more precise position estimation for itself and maintain a more stable flight.

**Reply by EJKT** [reply]
Hi Marios,

Thanks for the reply. I indeed figured out it may have something to do with too abrupt commands with little delay in between them. I forgot to mention this in the original question but I did observe that the readings of the position in the cfclient were appropriate. Good to know that the commander can be used with the HL one.

Thanks!

**Reply by marios-stam** [reply]
That is great, happy to help.
If you do not have anything else to add, could  you mark this as an answer by pressing 'mark as answer'?
